core/regions/src/main/java/software/amazon/awssdk/regions/internal/util/EC2MetadataUtils.java

```diff
@@ -587,4 +638,120 @@ public URI endpoint() {
             return requestHeaders;
         }
     }
+
+
```

Comment:
Optional : The comment for this class is missing as compared to the original one

Reply:
Added a comment